gc/gc-base/src/main/java/org/projectnessie/versioned/persist/gc/DistributedIdentifyUtil.java

```diff
@@ -0,0 +1,278 @@
+import java.util.Objects;
+import java.util.OptionalInt;
+import java.util.Set;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+import javax.validation.constraints.NotNull;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.sql.SparkSession;
+import org.projectnessie.client.NessieClientBuilder;
+import org.projectnessie.client.StreamingUtil;
+import org.projectnessie.client.api.NessieApiV1;
+import org.projectnessie.client.http.HttpClientBuilder;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.CommitMeta;
+import org.projectnessie.model.Content;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.LogResponse;
+import org.projectnessie.model.Operation;
+import org.projectnessie.model.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DistributedIdentifyUtil implements Serializable {
+
+  private static final long serialVersionUID = -5412153338307931076L;
+  private final SparkSession session;
+  private final Map<String, String> clientBuilderConfiguration;
+
+  public DistributedIdentifyUtil(
+      SparkSession session, Map<String, String> clientBuilderConfiguration) {
+    this.session = session;
+    this.clientBuilderConfiguration = clientBuilderConfiguration;
+  }
+
+  public static NessieApiV1 getApi(Map<String, String> configuration) {
+    // TODO: add this config - nessie.client-builder-impl
+    String clientBuilderClassName = configuration.get("nessie.client-builder-impl");
+    if (clientBuilderClassName == null) {
+      // Use default HttpClientBuilder
+      return HttpClientBuilder.builder().fromConfig(configuration::get).build(NessieApiV1.class);
+    }
+    // Use custom client builder
+    try {
+      NessieClientBuilder builder =
+          (NessieClientBuilder)
+              Class.forName(clientBuilderClassName).getDeclaredConstructor().newInstance();
+      return (NessieApiV1) builder.fromConfig(configuration::get).build(NessieApiV1.class);
+    } catch (ClassNotFoundException e) {
+      throw new IllegalArgumentException(
+          String.format("No custom client builder class found for '%s' ", clientBuilderClassName));
+    } catch (InvocationTargetException
+        | InstantiationException
+        | IllegalAccessException
+        | NoSuchMethodException e) {
+      throw new IllegalArgumentException(
+          String.format("Could not initialize '%s': ", clientBuilderClassName), e);
+    }
+  }
+
+  public Map<String, ContentBloomFilter> getLiveContents(
+      List<Reference> references,
+      Map<String, Instant> cutOffTimestampPerRef,
+      Instant defaultCutOffTimestamp) {
+
+    List<Map<String, ContentBloomFilter>> bloomFilterMaps =
+        new JavaSparkContext(session.sparkContext())
+            .parallelize(references, references.size())
```

Comment:
Is `references.size()` always correct here?

Reply:
it will launch `references.size()` number of tasks, finally the parallelism is based on the cores available.

Reply:
Is it efficient in Spark to do this? I thought this should be configurable?

Reply:
executor instance are always configurable, I will open a config for this with default value `references.size()`